cabal-install/src/Distribution/Client/IndexUtils.hs

```diff
@@ -275,17 +276,12 @@ getSourcePackagesAtIndexState verbosity repoCtxt mb_idxState mb_activeRepos = do
             return ()
         IndexStateTime ts0 -> do
             when (isiMaxTime isi /= ts0) $
-                if ts0 > isiMaxTime isi
-                    then warn verbosity $
-                                   "Requested index-state " ++ prettyShow ts0
-                                ++ " is newer than '" ++ unRepoName rname ++ "'!"
-                                ++ " Falling back to older state ("
-                                ++ prettyShow (isiMaxTime isi) ++ ")."
-                    else info verbosity $
-                                   "Requested index-state " ++ prettyShow ts0
-                                ++ " does not exist in '"++ unRepoName rname ++"'!"
-                                ++ " Falling back to older state ("
-                                ++ prettyShow (isiMaxTime isi) ++ ")."
+                info verbosity $
+                     "There is no index-state for '"
+                  ++ unRepoName rname ++ "' exactly at the requested timestamp ("
+                  ++ prettyShow ts0 ++ "). Falling back to the previous index-state that exists: "
+                  ++ prettyShow (isiMaxTime isi)
```

Comment:
I think this logic is wrong. It reads: "if the specified index-state is not equal to the latest index-state, then use the latest index-state".

This would trigger for all historical index-states, am I wrong?